### PR TITLE
Connnect colony stats on activity page and on colony home

### PR DESCRIPTION
**Pull request by thomassdk**
## Description

This PR connects the stats on the colony page (the number of active actions) and the stats on the activity page (Active actions, Recent actions and most active team)

[Screencast from 2023-11-28 10-11-28.webm](https://github.com/JoinColony/colonyCDapp/assets/32598350/20fc6e3c-06c6-4c75-bcd5-f3e3b35d5c63)

image

To test create actions and these pages should be populated with the results

**New stuff** ✨

There have been some fun opensearch aggregate queries added to the schema

### CODE REVIEW

**Review by arrenv — CHANGES_REQUESTED** [review]
@thomassdk 

1. How is this Recent actions value calculated? I am getting 42 in the widget, 40 in the activity list, the database is showing 49 actions.

Also, the date of actions is in the future in the activity feed and in the database, so it seems odd that it is counting it in the widget. With only 8 that fall in the past 30 days.

2. I have an motion the the staking phase, however, the active actions value is not showing any value on the dashboard and activity pages.

**Comment by thomassdk** [comment]
Thankyou for your review @arrenv 

1. recent actions values actions are calculated counting everything from 30 days ago as discussed. The issue with future dates seems like a non problem as putting in code to discount motions that happen in the future won't be an issue when the app is live no?  We can have a call to discuss this further.

2. We are counting all motions that are not finalized, perhaps this is wrong I will double check now

File: src/components/frame/v5/pages/ActivityPage/hooks.tsx [hunk 1]
```diff
@@ -1,17 +1,69 @@
...
+  const midnightThirtyDaysAgo = startOfDay(thirtyDaysAgo);
+  return midnightThirtyDaysAgo.toISOString();
+};
 
 export const useActivityFeedWidgets = (): WidthBoxItem[] => {
+  const { colony } = useColonyContext();
+  const { domains, colonyAddress = '' } = colony ?? {};
+  const { data: motionData } = useGetActiveMotionsQuery({
+    variables: {
+      colonyId: colonyAddress ?? '',
```
Comment (bassgeta) [inline src/components/frame/v5/pages/ActivityPage/hooks.tsx:26]: I guess we don't need ?? if we have a fallback here

File: src/components/frame/v5/pages/ActivityPage/hooks.tsx [hunk 2]
```diff
@@ -1,17 +1,69 @@
...
+    },
+  });
+
+  const activeActionsNumber = motionData?.searchColonyMotions?.total ?? 0;
+
+  const { data: actionData } = useGetTotalColonyActionsQuery({
+    variables: {
+      colonyId: colonyAddress ?? '',
```
Comment (bassgeta) [inline src/components/frame/v5/pages/ActivityPage/hooks.tsx:34]: same with the ?? operator here

File: src/components/frame/v5/pages/ActivityPage/hooks.tsx [hunk 3]
```diff
@@ -1,17 +1,69 @@
...
+      since: getThirtyDaysAgoIso(),
+    },
+  });
+
+  const recentActionsCount = actionData?.searchColonyActions?.total ?? 0;
+
+  const { data: domainData } = useGetTotalColonyDomainActionsQuery({
+    variables: {
+      colonyId: colonyAddress ?? '',
```
Comment (bassgeta) [inline src/components/frame/v5/pages/ActivityPage/hooks.tsx:43]: same with the ?? operator here

**Review by bassgeta — CHANGES_REQUESTED** [review]

File: src/components/frame/v5/pages/ActivityPage/hooks.tsx (hunk 1 above)
Comment (thomassdk) [reply src/components/frame/v5/pages/ActivityPage/hooks.tsx:26]: very good point lazy copy and paste strikes again :)

**Review by arrenv — DISMISSED** [review]
Thanks for the updates, stats are showing correctly from what I have tested.

I created two new motions, staked one past 10% and did not stake another, the count increased by one as expected. Nice work!

**Review by bassgeta — APPROVED** [review]

File: src/components/frame/v5/pages/ActivityPage/hooks.tsx [hunk 4]
```diff
@@ -1,25 +1,82 @@
...
+    variables: {
+      colonyId: colonyAddress,
+    },
+  });
+
+  const domainsActionCount =
+    // Buckets exists on this type but for some reason typescript is not
+    // recognising this
+    // @ts-ignore
+    domainData?.searchColonyActions?.aggregateItems[0]?.result?.buckets ?? [];
```
Comment (jakubcolony) [inline src/components/frame/v5/pages/ActivityPage/hooks.tsx:56]: I suspect it's the codegen that can't pick up union types without a separate plugin: https://the-guild.dev/graphql/codegen/plugins/other/fragment-matcher
It was pretty straightforward last time I set it up so I think it's worth getting sorted in this PR as we're gonna have the same problem with all GQL unions. Feel free to give me call if you need a hand with this

File: src/graphql/queries/actions.graphql [hunk 5]
```diff
@@ -43,12 +43,31 @@ query GetSafeTransactionStatus($input: GetSafeTransactionStatusInput!) {
...
+    filter: { colonyId: { eq: $colonyId } }
+    aggregates: { name: "FromDomainIdCount", type: terms, field: fromDomainId }
+  ) {
+    aggregateItems {
+      result {
+        ... on SearchableAggregateBucketResult {
+          __typename
+          buckets {
+            key
+            doc_count
```
Comment (jakubcolony) [inline src/graphql/queries/actions.graphql:63]: Can we rename this to be camelCase? Should be able to do 
```gql
docCount: doc_count
```

**Review by jakubcolony — CHANGES_REQUESTED** [review]

File: src/components/frame/v5/pages/ActivityPage/hooks.tsx (hunk 4 above)
Comment (thomassdk) [reply src/components/frame/v5/pages/ActivityPage/hooks.tsx:56]: turns out I wasn't understanding this issue clearly enough as the type was generated correctly have fixed this issue now :)

**Review by melyndav — APPROVED** [review]
Nice work @thomassdk! Get this baby merged! ;)

**Review by arrenv — APPROVED** [review]
Good to go!

**Review by jakubcolony — APPROVED** [review]
LGTM, nice work! 💪

**Pull request merged**